### PR TITLE
AIK-3167 : Switch the python firewall agent to an IPC socket

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/agent/__init__.py [hunk 1]
```diff
@@ -3,72 +3,110 @@
...
-import queue
+import os
+import multiprocessing.connection as con
+from multiprocessing import Process
 from threading import Thread
+from queue import Queue
 from aikido_firewall.helpers.logging import logger
 
-AGENT_SEC_INTERVAL = 60
+AGENT_SEC_INTERVAL = 5
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py]: 600

File: aikido_firewall/agent/__init__.py [hunk 2]
```diff
@@ -3,72 +3,110 @@
...
+IPC_ADDRESS = ("localhost", 9898)  # Specify the IP address and port
 
 
-class AikidoThread:
+class AikidoProc:
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py]: aikidoAgentProcess

File: aikido_firewall/agent/__init__.py [hunk 3]
```diff
@@ -3,72 +3,110 @@
...
+        # Start reporting thread :
+        Thread(target=self.reporting_thread).start()
+
+        while True:
+            conn = listener.accept()
+            logger.debug("connection accepted from %s", listener.last_accepted)
+            while True:
+                data = conn.recv()
+                logger.error(data)  # Temporary debugging
+                if data[0] == "SQL_INJECTION":
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py]: attack

File: aikido_firewall/agent/__init__.py [hunk 4]
```diff
@@ -3,72 +3,110 @@
...
+def start_ipc():
     """
     Starts a thread to handle incoming/outgoing data
     """
     # pylint: disable=global-statement # We need this to be global
-    global agent
+    global ipc
+    aikido_secret_key_env = os.getenv("AIKIDO_SECRET_KEY")
+    if not aikido_secret_key_env:
+        raise EnvironmentError("AIKIDO_SECRET_KEY is not set.")
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py]: lets create a key ourselves and use that

Comment (bitterpanda63) [reply aikido_firewall/agent/__init__.py]: Done

File: aikido_firewall/agent/__init__.py (hunk 3 above)
Comment (bitterpanda63) [reply aikido_firewall/agent/__init__.py]: Done

File: aikido_firewall/background_process/__init__.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+        while True:
+            conn = listener.accept()
+            logger.debug("connection accepted from %s", listener.last_accepted)
+            while True:
+                data = conn.recv()
+                logger.error(data)  # Temporary debugging
+                if data[0] == "ATTACK":
+                    self.queue.put(data[1])
+                elif data[0] == "CLOSE":
```
Comment (willem-delbare) [inline aikido_firewall/background_process/__init__.py]:
```suggestion
                elif data[0] == "CLOSE": // this is a kind of EOL for python IPC
```

File: aikido_firewall/background_process/__init__.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+        while True:
+            conn = listener.accept()
+            logger.debug("connection accepted from %s", listener.last_accepted)
+            while True:
+                data = conn.recv()
+                logger.error(data)  # Temporary debugging
+                if data[0] == "ATTACK":
+                    self.queue.put(data[1])
+                elif data[0] == "CLOSE": // this is a kind of EOL for python IPC
```
Comment (willem-delbare) [inline aikido_firewall/background_process/__init__.py]:
```suggestion
                elif data[0] == "CLOSE": # this is a kind of EOL for python IPC
```

File: aikido_firewall/background_process/__init__.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+
+def start_background_process():
+    """
+    Starts a process to handle incoming/outgoing data
+    """
+    # pylint: disable=global-statement # We need this to be global
+    global ipc
+    # Generate a secret key :
+    generated_key_bytes = secrets.token_bytes(32)
```
Comment (willem-delbare) [inline aikido_firewall/background_process/__init__.py:83]: aikido token

**Pull request merged**